RenderImage::hasNonBitmapImage()/hasAnimatedImage() is unused

kkinnunen-apple opened the pull request:
#### 3393aa184c13fbd6785d0320ca8454572efc78c9
<pre>
RenderImage::hasNonBitmapImage()/hasAnimatedImage() is unused
<a href="https://bugs.webkit.org/show_bug.cgi?id=286418">https://bugs.webkit.org/show_bug.cgi?id=286418</a>
<a href="https://rdar.apple.com/143476837">rdar://143476837</a>

Reviewed by Anne van Kesteren.

Remove the unused functions. Having unused casts and functions make
the Image interface modification harder.

* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::isMultiRepresentationHEIC const):
(WebCore::RenderImage::hasNonBitmapImage const): Deleted.
(WebCore::RenderImage::hasAnimatedImage const): Deleted.
* Source/WebCore/rendering/RenderImage.h:

Canonical link: <a href="https://commits.webkit.org/290944@main">https://commits.webkit.org/290944@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b5bc2a6f54492ec81f8a748a70c2d43b87f9effd

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/91530 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/11061 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/591 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/96499 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/42216 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/93580 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/11439 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/19472 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/70287 "Failure limit exceed. At least found 1 new test failure: imported/w3c/web-platform-tests/fullscreen/api/document-exit-fullscreen-nested.html (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/27806 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/94531 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/8732 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/82924 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50611 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/8496 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/512 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/41386 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/78816 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/516 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/98502 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/18691 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/13764 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/79309 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/18946 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/78762 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/78512 "Found 1 new API test failure: /WebKitGTK/TestInspector:/webkit/WebKitWebInspector/manual-attach-detach (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19418 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/23036 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/392 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/11815 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/18687 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/23963 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/18398 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/21858 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/20164 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/rendering/RenderImage.cpp:
```diff
@@ -461,30 +461,14 @@ bool RenderImage::shouldDisplayBrokenImageIcon() const
         return false;
 
-    return imageElement->isMultiRepresentationHEIC();
+    return imageElement-><isMultiRepresentationHEIC>();
```
I don't understand this change. Otherwise this would be an easy r+.

kkinnunen-apple committed:
RenderImage::hasNonBitmapImage()/hasAnimatedImage() is unused
https://bugs.webkit.org/show_bug.cgi?id=286418
rdar://143476837

Reviewed by Anne van Kesteren.

Remove the unused functions. Having unused casts and functions make
the Image interface modification harder.

* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::isMultiRepresentationHEIC const):
(WebCore::RenderImage::hasNonBitmapImage const): Deleted.
(WebCore::RenderImage::hasAnimatedImage const): Deleted.
* Source/WebCore/rendering/RenderImage.h:

Canonical link: https://commits.webkit.org/290944@main
